sumologic/resource_sumologic_kinesis_log_source_test.go

```diff
@@ -0,0 +1,203 @@
+        if rs.Primary.ID == "" {
+            return fmt.Errorf("KinesisLog Source ID is not set")
+        }
+        id, err := strconv.Atoi(rs.Primary.ID)
+        if err != nil {
+            return fmt.Errorf("KinesisLog Source id should be int; got %s", rs.Primary.ID)
+        }
+        collectorID, err := strconv.Atoi(rs.Primary.Attributes["collector_id"])
+        if err != nil {
+            return fmt.Errorf("Encountered an error: " + err.Error())
+        }
+        c := testAccProvider.Meta().(*Client)
+        kinesisLogSourceResp, err := c.GetKinesisLogSource(collectorID, id)
+        if err != nil {
+            return err
+        }
+        *kinesisLogSource = *kinesisLogSourceResp
+        return nil
+    }
+}
+func testAccCheckKinesisLogSourceValues(kinesisLogSource *KinesisLogSource, name, description, category string) resource.TestCheckFunc {
+    return func(s *terraform.State) error {
+        if kinesisLogSource.Name != name {
+            return fmt.Errorf("bad name, expected \"%s\", got: %#v", name, kinesisLogSource.Name)
+        }
+        if kinesisLogSource.Description != description {
+            return fmt.Errorf("bad description, expected \"%s\", got: %#v", description, kinesisLogSource.Description)
+        }
+        if kinesisLogSource.Category != category {
+            return fmt.Errorf("bad category, expected \"%s\", got: %#v", category, kinesisLogSource.Category)
+        }
+        return nil
+    }
+}
+func testAccSumologicKinesisLogSourceConfig(cName, cDescription, cCategory, sName, sDescription, sCategory, testAwsRoleArn, testAwsBucket string) string {
+    return fmt.Sprintf(`
+resource "sumologic_collector" "test" {
+    name = "%s"
+    description = "%s"
+    category = "%s"
+}
+
+resource "sumologic_kinesis_log_source" "kinesisLog" {
+    name = "%s"
+    description = "%s"
+    category = "%s"
+    content_type  = "KinesisLog"
+    collector_id = "${sumologic_collector.test.id}"
+    authentication {
+        type = "AWSRoleBasedAuthentication"
+        role_arn = "%s"
+    }
+    path {
+        type = "KinesisLogPath"
+        bucket_name     = "%s"
+        path_expression = "http-endpoint-failed/*"
+        scan_interval   = 30000
+    }
+}
+`, cName, cDescription, cCategory, sName, sDescription, sCategory, testAwsRoleArn, testAwsBucket)
+}
+
+func testAccSumologicKinesisLogSourceNoAuthConfig(cName, cDescription, cCategory, sName, sDescription, sCategory string) string {
+	return fmt.Sprintf(`
+resource "sumologic_collector" "test" {
+    name = "%s"
+    description = "%s"
+    category = "%s"
+}
+
+resource "sumologic_kinesis_log_source" "kinesisLog" {
+    name = "%s"
+    description = "%s"
+    category = "%s"
+    content_type  = "KinesisLog"
+    collector_id = "${sumologic_collector.test.id}"
+    authentication {
+        type = "NoAuthentication"
+    }
+    path {
+        type = "NoPathExpression"
+    }
+}
+`, cName, cDescription, cCategory, sName, sDescription, sCategory)
```

Comment:
I think this block of code is still missing one tab indentation 😓

Reply:
For this block, it's actually intended to format this way because it's a multiple string passed into `fmt.Sprintf()`. This is not part of the code and adding the indentation here will mess up the multiline string format itself. So we should leave it as this.